sui_core/src/authority/authority_store.rs

```diff
@@ -17,20 +18,56 @@ use typed_store::rocks::{DBBatch, DBMap, TypedStoreError};
 
 use typed_store::{reopen, traits::Map};
 
-pub type AuthorityStore = SuiDataStore<false>;
+pub type AuthorityStore = SuiDataStore<true>;
 #[allow(dead_code)]
-pub type ReplicaStore = SuiDataStore<true>;
+pub type ReplicaStore = SuiDataStore<false>;
 
 const NUM_SHARDS: usize = 4096;
 
 /// The key where the latest consensus index is stored in the database.
 // TODO: Make a single table (e.g., called `variables`) storing all our lonely variables in one place.
 const LAST_CONSENSUS_INDEX_ADDR: u64 = 0;
 
+#[derive(Serialize, Deserialize)]
+pub enum RawOrSignedData<D, const SIGNED: bool> {
+    RawData(D),
+    SignedData(AuthoritySignedData<D>),
+}
+
+impl<D> From<RawOrSignedData<D, true>> for AuthoritySignedData<D> {
+    fn from(other: RawOrSignedData<D, true>) -> AuthoritySignedData<D> {
+        match other {
+            RawOrSignedData::RawData(_) => panic!("Data is signed"),
+            RawOrSignedData::SignedData(s) => s,
+        }
+    }
+}
+
+impl<D> From<AuthoritySignedData<D>> for RawOrSignedData<D, true> {
+    fn from(other: AuthoritySignedData<D>) -> RawOrSignedData<D, true> {
+        RawOrSignedData::SignedData(other)
+    }
+}
+
+impl<D> RawOrSignedData<D, false> {
+    pub fn into(other: RawOrSignedData<D, false>) -> D {
+        match other {
+            RawOrSignedData::RawData(r) => r,
+            RawOrSignedData::SignedData(_) => panic!("Data is not signed"),
```

Comment:
Same here.  It would be much more ergonomic to implement `TryFrom<RawOrSignedData<...>>` for D, which can return errors.